mlos_bench/mlos_bench/optimizers/base_optimizer.py

```diff
@@ -347,13 +347,13 @@ def not_converged(self) -> bool:
         return self._iter < self._max_iter
 
     @abstractmethod
-    def get_best_observation(self) -> Union[Tuple[float, TunableGroups], Tuple[None, None]]:
+    def get_best_observation(self) -> Union[Tuple[Dict[str, float], TunableGroups], Tuple[None, None]]:
```

Comment:
```suggestion
    def get_best_observations(self) -> Union[Tuple[Dict[str, float], TunableGroups], Tuple[None, None]]:
```

Don't we need to return the full pareto frontier now?

Reply:
Another option is we extend it to `get_best_observation(weights={
  "opt_target_1": 1,
  "opt_target_2": 0,
  "opt_target_3": 1,
})

As a means to flatten the parent frontier into a single point by means of multi-dimensional score weighting.

Reply:
I would do it in the next PR after we add multi-target optimization to `mlos_core` - rn I am not even sure that any of our optimizers can return all points of the Pareto frontier. (also, it will require a lot of changes to our unit tests)